secure-api-gateway-ob-uk-rs-server/src/main/java/com/forgerock/sapi/gateway/ob/uk/rs/server/api/obie/payment/v4_0_0/vrp/DomesticVrpConsentsApiController.java

```diff
@@ -0,0 +1,202 @@
+import uk.org.openbanking.datamodel.v4.vrp.OBPAFundsAvailableResult1;
+import uk.org.openbanking.datamodel.v4.vrp.OBPAFundsAvailableResult1FundsAvailable;
+import uk.org.openbanking.datamodel.v4.vrp.OBVRPFundsConfirmationRequest;
+import uk.org.openbanking.datamodel.v4.vrp.OBVRPFundsConfirmationResponse;
+import uk.org.openbanking.datamodel.v4.vrp.OBVRPFundsConfirmationResponseData;
+
+@jakarta.annotation.Generated(value = "org.openapitools.codegen.languages.SpringCodegen")
+@Controller("DomesticVrpConsentApiV4.0.0")
+@Slf4j
+public class DomesticVrpConsentsApiController implements DomesticVrpConsentsApi {
+
+    private final FundsAvailabilityService fundsAvailabilityService;
+
+    private final DomesticVRPConsentStoreClient consentStoreClient;
+
+    private final OBValidationService<OBDomesticVRPConsentRequest> vrpConsentValidator;
+
+    private final OBValidationService<VRPFundsConfirmationValidationContext> vrpFundsConfirmationValidator;
+
+    private final OBDomesticVRPConsentResponseFactory responseFactory;
+
+    public DomesticVrpConsentsApiController(FundsAvailabilityService fundsAvailabilityService,
+                                            @Qualifier("v4.0.0RestDomesticVRPConsentStoreClient")DomesticVRPConsentStoreClient consentStoreClient,
+                                            OBValidationService<OBDomesticVRPConsentRequest> vrpConsentValidator,
+                                            OBValidationService<VRPFundsConfirmationValidationContext> vrpFundsConfirmationValidator,
+                                            OBDomesticVRPConsentResponseFactory responseFactory) {
+        this.fundsAvailabilityService = fundsAvailabilityService;
+        this.vrpConsentValidator = vrpConsentValidator;
+        this.vrpFundsConfirmationValidator = vrpFundsConfirmationValidator;
+        this.consentStoreClient = consentStoreClient;
+        this.responseFactory = responseFactory;
+    }
+
+    @Override
+    public ResponseEntity<OBDomesticVRPConsentResponse> domesticVrpConsentsGet(String consentId,
+                                                                               String authorization,
+                                                                               String xFapiAuthDate,
+                                                                               String xFapiCustomerIpAddress,
+                                                                               String xFapiInteractionId,
+                                                                               String xCustomerUserAgent,
+                                                                               String apiClientId,
+                                                                               HttpServletRequest request) throws OBErrorResponseException {
+
+        log.info("domesticVrpConsentsGet - consentId: {}, apiClientId: {}, x-fapi-interaction-id: {}", consentId, apiClientId, xFapiInteractionId);
```

Comment:
Info-level logging necessary here?

Won't the logs be a bit noisy (and expensive)? Maybe trace?

Reply:
And elsewhere (below).

Reply:
Changed logging level